website/blog/2024-04-22-0.74-yoga-improvements-preparing-for-the-new-architecture.md

```diff
@@ -0,0 +1,412 @@
+
+### Breaking Changes
+
+- [Breaking Changes](/blog/2024/04/22/0.74-yoga-improvements-preparing-for-the-new-architecture#breaking-changes)
+
+<!--truncate-->
+
+## Highlights
+
+### Yoga Improvements
+
+#### New Layout Behaviors
+
+React Native 0.74 includes [Yoga 3.0](https://yogalayout.dev/blog/announcing-yoga-3.0), the newest version of our layout engine, which enables more correct layout behavior. These changes make styling more predictable, and enable rendering components written for the web with greater fidelity.
+
+React Native intentionally preserves some incorrect layout behaviors, where fixing them was found to effect a signigicant number of real-world components. Layout conformance will be able to be configured more granularly in future versions of React Native.
+
+:::warning
+
+React Native [previously flipped](https://yogalayout.dev/blog/announcing-yoga-3.0#correct-handling-of-logical-edges-in-row-reverse-containers) `left`/`right` (and `start`/`end`) edges when dealing with `margin`, `padding`, or `border`, set on a `row-reverse` container. Code which previously relied on edges being inverted may need to be updated to continue rendering correctly.
+
+<table>
+<tr>
+  <th>Style</th>
+  <th>Before</th>
+  <th>After</th>
+</tr>
+<tr>
+<td width="350">
+
+```jsx
+<View
+  style={{
+    flexDirection: 'row',
+    backgroundColor: 'red',
+    margin: 10,
+    width: 200,
+    height: 100,
+  }}>
+  <View
+    style={{
+      flexDirection: 'row-reverse',
+      backgroundColor: 'blue',
+      flex: 1,
+      marginLeft: 50,
+    }}>
+    <View
+      style={{
+        backgroundColor: 'green',
+        height: '50%',
+        flex: 1,
+        marginLeft: 50,
+      }}
+    />
+  </View>
+</View>
+```
+
+</td>
+<td>
+
+![Previous layout](../static/blog/assets/0.74-row-reverse-before.png)
+
+</td>
+<td>
+
+![New layout](../static/blog/assets/0.74-row-reverse-after.png)
+
+</td>
+</tr>
+</table>
+
+:::
+
+#### Support for `align-content: 'space-evenly'`
+
+[`alignContent: 'space-evenly'`](https://developer.mozilla.org/en-US/docs/Web/CSS/align-content#space-evenly) distributes the lines in a multi-line flex container using evenly spaced gaps, placed between line and container edges.
+
+<figure>
+  <img alt="Visual reference for alignContent behaviours" src="/blog/assets/0.74-align-content.png" width="360" />
+  <figcaption><small>Source: [World Wide Web Consortium](https://www.w3.org/TR/css-align-3/#distribution-values)</small></figcaption>
+</figure>
+
+#### Support for `position: 'static'` (New Architecture)
+
+Elements marked as [`position: 'static'`](https://developer.mozilla.org/en-US/docs/Web/CSS/position#static) may not be offset, and are not considered when determining the [containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block) of an absolutely positioned element. This allows positioning an element relative to an ancestor which is not its direct parent.
```

Comment:
so... did we explicitly decide to keep this New Architecture bit in the blogpost? Could I ask what's the rationale for this choice?

It might feel confusing for the users that we have:
- New Arch docs in the working group
- New Arch announcements in the working group
- This specific New Arch callout in the Blogpost

Reply:
It's a soft decision from me — happy to be overridden with more context. However:
- This is a small section that's logically tied to the "Yoga improvements" theme.
- I think there's some value in highlighting, over time, that a subset of useful new features will be New Arch only.

Reply:
I believe it makes sense to keep it here as we do have a collection of Yoga updates. I wouldn't scatter them between WG and blogpost, it makes more sense to have all of them here